mithril-common/src/crypto_helper/merkle_map.rs

```diff
@@ -132,22 +157,33 @@ impl<K: MKMapKey, V: MKMapValue<K>> MKMap<K, V> {
         if leaves.is_empty() {
             return Err(anyhow!("MKMap could not compute proof for empty leaves"));
         }
-        let leaves_by_keys: HashMap<K, Vec<MKTreeNode>> = leaves
+
+        let can_compute_proof_map: HashMap<K, V> = self
```

Comment:
This function is a bit complicated for a recursive function. 
We could extract some parts to reduce its size.
It can also help to test and show the behavior of sub parts of this function

Reply:
I created a new function to compute the grouped leaves by keys to make the `compute_proof` function easier to read